.github/workflows/pypi-publish.yaml

```diff
@@ -1,35 +1,71 @@
 name: Publish Python Package
+# Upload to PyPI is triggered by creating & publishing a release in GitHub UI
 
 on:
   release:
+    # Trigger the workflow only for real releases but no for draft releases
     types: [created]
 
 jobs:
-  build-n-publish:
-    name: Build and publish Python 🐍 distributions 📦 to PyPI
+  build:
+    name: Build Python 🐍 distributions 📦 for publishing to PyPI
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4.2
+      # By default the ref (branch, tag or SHA) that triggered the workflow,
+      # is checked out. So the version tag is available for dynamic versioning
+      # and the option "with: fetch-depth: 0" is not needed here.
 
     - name: Set up Python
-      uses: actions/setup-python@v2.2.2
+      uses: actions/setup-python@v5.5
       with:
         python-version: 3.12
 
     - name: Install Poetry
-      run: pipx install poetry
-
-    - name: Install dependencies
-      run: poetry install --no-interaction
+      run: |
+        pipx install poetry
+        pipx inject poetry poetry-dynamic-versioning
 
     - name: Build source and wheel archives
       run: |
-        poetry version $(git describe --tags --abbrev=0)
         poetry build
+        # Check if correct tag is available
+        git describe --tags --abbrev=0
 
-    - name: Publish distribution 📦 to PyPI
-      uses: pypa/gh-action-pypi-publish@v1.2.2
+    - name: Store built distribution
+      uses: actions/upload-artifact@v4.6
       with:
-        user: __token__
-        password: ${{ secrets.pypi_password }}
+        name: distribution-files
+        path: dist/
+
+  pypi-publish:
+    name: Build and publish Python 🐍 package 📦 to PyPI
+    needs: build
+    runs-on: ubuntu-latest
+    # Next 5 lines prepare for trusted publishing: https://docs.pypi.org/trusted-publishers/adding-a-publisher/
+    # environment:
+    #   name: pypi-release
+    #   url: https://pypi.org/p/linkml-runtime
+    # permissions:
+    #   id-token: write  # this permission is mandatory for trusted publishing
+    steps:
+      - name: Download built distribution
+        uses: actions/download-artifact@v4.2
+        with:
+          name: distribution-files
+          path: dist
+
+      - name: Publish package 📦 to PyPI
+        if: github.event_name == 'release'
+        uses: pypa/gh-action-pypi-publish@v1.12
+        with:
+          user: __token__
+          password: ${{ secrets.pypi_password }}
+          # verbose: true
+        
+# Used actions: (updates managed by dependabot)
+# - https://github.com/actions/checkout
+# - https://github.com/actions/setup-python
+# - https://github.com/actions/upload-artifact
+# - https://github.com/actions/download-artifact
+# - https://github.com/pypa/gh-action-pypi-publish/
```

Comment:
1. Some files have this list at the top, some at the bottom. Can we standardise on a location?

2. Is this list useful if deps are managed by dependabot anyway?

Reply:
It is only useful for other developers. Dependabot does not need it. Maybe we should remove this info everywhere?